packages/create-sitecore-jss/src/templates/nextjs-personalize/src/lib/pos-resolver.ts

```diff
@@ -1,50 +1,9 @@
-// This class is used to get point of sale identifier(s). Resolving from env variable, but it can be expanded or change in future if needed.
-export class PosResolver {
-  /**
-   * Resolve point of sale by locale
-   * @param {string} locale
-   * @returns {string}
-   */
-  static resolve = (locale: string): string => {
-    try {
-      // POS can be multi-valued (one entry per locale) or single valued so we parse it
-      // POS must be valid in order to save events (domain name might be taken but it must be defined in CDP settings)
-      const parsedPos = PosResolver.parseEnvValue(process.env.NEXT_PUBLIC_CDP_POINTOFSALE);
-      if (typeof parsedPos == 'string') return parsedPos;
-      else return parsedPos[locale];
-    } catch (error) {
-      console.log(error);
-      return '';
-    }
-  };
+import { PosResolver } from '@sitecore-jss/sitecore-jss-nextjs/middleware';
+import { siteResolver } from './site-resolver';
 
-  /**
-   * Entry method to parse multi or single value env variables
-   * @param {string} envValue - can be undefined when providing values via process.env
-   * @returns Record or string depending on input format
-   */
-  private static parseEnvValue = (envValue?: string): Record<string, string> | string => {
-    if (!envValue) {
-      return '';
-    }
-    if (envValue.startsWith('{') && envValue.endsWith('}')) {
-      return PosResolver.parseMultiValueAsRecord(envValue);
-    } else return envValue;
-  };
+/*
+  The pos resolver stores sites point of sale
+*/
 
-  /**
-   * Parses raw env variable value into a Record (locale-bound in some cases)
-   * @param {string} mvEnv input
-   * @returns non-empty Record
-   * @throws error when input JSON can't be parsed
-   */
-  private static parseMultiValueAsRecord = (mvEnv: string): Record<string, string> => {
-    try {
-      return JSON.parse(mvEnv) as Record<string, string>;
-    } catch (error) {
-      console.log('Parsing of multivalue env variable failed');
-      console.log(`Attempted to parse ${mvEnv}`);
-      throw error;
-    }
-  };
-}
+/** PosResolver singleton */
+export const posResolver = new PosResolver({ getSite: siteResolver.getByName });
```

Comment:
Hmm.. I know we discussed but I'm wondering again about the need for a POS resolver (both the class in base package and singleton here). We now have everything we need on the `siteResolver`/`SiteInfo`. If we have `SiteInfo`, we can determine POS with:
```
site.pointOfSale ? site.pointOfSale[locale] || site.pointOfSale[site.language] : ''
```

Thinking about the 2 cases we use POS resolver:

1. `PersonalizeMiddleware`: We have `getSite` which returns `SiteInfo` (should be good, can remove `getPointOfSale`)
2. `CdpPageView.tsx`: We have context `site`, can use `siteResolver.getByName` to return `SiteInfo`

Reply:
Makes sense to me. Initially I did similar approach but was worrying about duplicate logic regarding pos resolution, but I think not a big deal to apply that